### PR TITLE
feat: migrate code from googleapis/python-container

### PR DESCRIPTION
**Pull request by donmccasland**
- feat!: migrate to microgenerator (#33)
- docs(samples): add usage samples to show handling of LRO response Operation (#191)
- chore: Adding support for pytest-xdist and pytest-parallel (#204)
- chore(deps): update all dependencies (#206)
- chore(deps): update dependency google-cloud-container to v2.10.6 (#207)
- chore(deps): update dependency pytest to v7.1.0 (#209)
- chore(deps): update dependency pytest to v7.1.1 (#211)
- fix: test cleanup stages with try finally (#212)
- chore(deps): update dependency google-cloud-container to v2.10.7 (#215)
- chore(python): use black==22.3.0 (#217)
- chore(python): add nox session to sort python imports (#229)
- chore(deps): update dependency pytest to v7.1.2 (#232)
- chore(deps): update dependency backoff to v2 (#233)
- chore(deps): update dependency backoff to v2.0.1 (#235)
- fix: require python 3.7+ (#266)
- chore(deps): update all dependencies (#258)
- chore(deps): update all dependencies (#271)
- chore(deps): update all dependencies to v2.11.1 (#276)
- chore(deps): update dependency google-cloud-container to v2.11.2 (#280)
- chore(deps): update dependency pytest to v7.1.3 (#291)
- chore: detect samples tests in nested directories (#295)
- chore(deps): update dependency google-cloud-container to v2.12.0 (#299)
- chore(deps): update dependency google-cloud-container to v2.12.1 (#303)
- chore(deps): update dependency backoff to v2.2.1 (#304)
- chore(deps): update dependency google-cloud-container to v2.12.2 (#308)
- chore(deps): update dependency pytest to v7.2.0 (#310)
- chore(deps): update dependency google-cloud-container to v2.13.0 (#311)

## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: container/snippets/noxfile.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    ret["GOOGLE_CLOUD_PROJECT"] = os.environ[env_key]
+
+    # Apply user supplied envs.
+    ret.update(TEST_CONFIG["envs"])
+    return ret
+
+
+# DO NOT EDIT - automatically generated.
+# All versions used to test samples.
+ALL_VERSIONS = ["3.7", "3.8", "3.9", "3.10"]
```
Comment (nicain) [inline container/snippets/noxfile.py:91]:
```suggestion
ALL_VERSIONS = ["2.7","3.7", "3.8", "3.9", "3.10"]
```

File: container/AUTHORING_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1 @@
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md
```
Comment (dandhlee) [inline container/AUTHORING_GUIDE.md:1]:
```suggestion
See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md
```

File: container/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1 @@
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/CONTRIBUTING.md
```
Comment (dandhlee) [inline container/CONTRIBUTING.md:1]:
```suggestion
See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/CONTRIBUTING.md
```

File: container/snippets/noxfile.py [hunk 4]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2022 Google LLC
```
Comment (dandhlee) [inline container/snippets/noxfile.py:1]: Remove this file. Only `noxfile_config.py` is needed.

**Review by dandhlee — APPROVED** [review]
LGTM, not sure what could be wrong with the test. Could you try deleting `noxfile.py` and we can see?

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -28,6 +28,7 @@
 /codelabs/**/*                         @GoogleCloudPlatform/python-samples-reviewers
 /composer/**/*                         @leahecole @rachael-ds @rafalbiegacz @GoogleCloudPlatform/python-samples-reviewers
 /compute/**/*                          @m-strzelczyk @GoogleCloudPlatform/python-samples-reviewers
+/container/**/*                        @GoogleCloudPlatform/dee-platform-ops @GoogleCloudPlatform/python-samples-reviewers
```
Comment (dandhlee) [inline .github/CODEOWNERS:31]: Could you also update https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml?

**Review by dandhlee — CHANGES_REQUESTED** [review]

**Review by dandhlee — APPROVED** [review]
LGTM. Pending test passes.

**Comment by dandhlee** [comment]
Verified all tests related to container are passing, all other checklist items are complete! Merging with `Rebase and merge`.

**Pull request merged**